scipy/signal/_max_len_seq_inner.pyx

```diff
@@ -0,0 +1,31 @@
+# Author: Eric Larson
+# 2014
+
+import numpy as np
+cimport numpy as np
+cimport cython
+
+
+# Fast inner loop of max_len_seq.
+@cython.cdivision(True)  # faster modulo
+@cython.boundscheck(False)  # designed to stay within bounds
+@cython.wraparound(False)  # we don't use negative indexing
+def _max_len_seq_inner(np.ndarray[Py_ssize_t, ndim=1, mode='c'] taps,
+                       np.ndarray[np.int8_t, ndim=1, mode='c'] state,
+                       Py_ssize_t nbits, Py_ssize_t length,
+                       np.ndarray[np.int8_t, ndim=1, mode='c'] seq):
+    # Here we compute MLS using a shift register, indexed using a ring buffer
+    # technique (faster than using something like np.roll to shift)
+    cdef Py_ssize_t n_taps = taps.shape[0]
```

Comment:
I changed this to `shape[0]` because Cython knows what that means. Getting the `size` involves a conversion from C integer to Python and back.